### PR TITLE
Introduce .ko.yaml for configuration

### PR DESCRIPTION
**Pull request by mattmoor**
Add support for overriding both the default base image, as well as the base images for specific import paths.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -55,6 +55,12 @@
   revision = "47565b4f722fb6ceae66b95f853feed578a4a51c"
   version = "v0.3.3"
 
+[[projects]]
```
Comment (jonjohnsonjr) [inline Gopkg.lock]: AFAICT this isn't used?

File: cmd/ko/config.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+func GetMountPaths() []name.Repository {
+	repos := make([]name.Repository, 0, len(baseImageOverrides)+1)
+	repos = append(repos, defaultBaseImage.Context())
+	for _, v := range baseImageOverrides {
+		repos = append(repos, v.Context())
+	}
+	return repos
+}
+
+func init() {
```
Comment (jonjohnsonjr) [inline cmd/ko/config.go]: Per #74 can you add some context to these error messages?

File: cmd/ko/README.md [hunk 3]
```diff
@@ -166,6 +166,54 @@ to whatever `kubectl` context is active.
...
+you will want to override `ko`'s default behavior. This is done via `.ko.yaml`.
+
+`.ko.yaml` is put into the directory from which `ko` will be invoked.  If it
+is not present, then `ko` will rely on its default behaviors.
+
+### Overriding the default base image
+
+By default, `ko` makes use of `gcr.io/distroless/base:latest` as the base image
+for containers.  There are a wide array of scenarios in which overriding this
+makes sense, to name a few:
```
Comment (jonjohnsonjr) [inline cmd/ko/README.md]: nit: two isn't quite "a few" in my mind, I'd replace that with "for example:"

File: cmd/ko/README.md [hunk 4]
```diff
@@ -166,6 +166,54 @@ to whatever `kubectl` context is active.
...
+1. Pinning to a particular digest of this image for repeatable builds,
+1. Replacing this streamlined base image with another with better debugging
```
Comment (jonjohnsonjr) [inline cmd/ko/README.md]: Maybe link to https://github.com/GoogleContainerTools/distroless#debug-images

File: cmd/ko/README.md [hunk 5]
```diff
@@ -166,6 +166,54 @@ to whatever `kubectl` context is active.
...
+line to `.ko.yaml`:
+
+```yaml
+defaultBaseImage: gcr.io/another-project/another-image@sha256:deadbeef
+```
+
+### Overriding the base for particular imports
+
+Some of your binaries may have requirements that are a more unique, and you
+want to direct `ko` to use a particular base image for just those binaries.
```
Comment (jonjohnsonjr) [inline cmd/ko/README.md]: may want

File: cmd/ko/README.md (hunk 3 above)
Comment (mattmoor) [reply cmd/ko/README.md]: Done

File: cmd/ko/README.md (hunk 4 above)
Comment (mattmoor) [reply cmd/ko/README.md]: I was thinking ubuntu

File: cmd/ko/README.md (hunk 5 above)
Comment (mattmoor) [reply cmd/ko/README.md]: Done

File: cmd/ko/config.go (hunk 2 above)
Comment (mattmoor) [reply cmd/ko/config.go]: I did a pass to do this all over cmd/ko.

**Pull request merged**